tools/cli/lib/parseargs.c

```diff
@@ -21,6 +21,22 @@
 
 #include "includes.h"
 
+typedef struct _SetOptArgs
+{
+    TDNF_CMDOPT_TYPE Type;
+    const char *OptName;
+    const char *OptVal;
+} SetOptArgs;
+
+static SetOptArgs OptValTable[] = {
+    {CMDOPT_KEYVALUE, "sec-severity", NULL},
+    {CMDOPT_ENABLEREPO, "enablerepo", NULL},
+    {CMDOPT_DISABLEREPO, "disablerepo", NULL},
+    {CMDOPT_ENABLEPLUGIN, "enableplugin", NULL},
+    {CMDOPT_DISABLEPLUGIN, "disableplugin", NULL},
+    {CMDOPT_KEYVALUE, "skipconflicts skipobsoletes skipsignature skipdigest noplugins reboot-required security", "1"}
```

Comment:
if you add each with a delimiter (space okay) including the last one, you can do strcasecmp after strcat the delimiter to the search input. avoid loop

Reply:
Didn't get your point, please explain :)

Reply:
ah i see i meant strstr and wrote strcasecmp. 
this is what i meant for eg.
suppose CMDOPT_KEYVALUE, "skipconflicts;skipobsoletes;"
then, instead of strsep and loop to match, you could
say you have to_match="skipobsoletes", make delimited_tomatch="skipobsoletes;"
then you can do if (strstr("skipconflicts;skipobsoletes;", "skipobsoletes;")) add set opt values
no need for strsep and loop.

Reply:
I thought of it but it's not robust enough. If my input is:
tdnf check --blablabla-skipobsoletes, strstr will accept it :) that's why I went for the hard way of using strsep :)

Reply:
Or I got confused with haystack needle of strstr, time to check man page :)

Reply:
ParseOption is not called with raw input. if you check the calls, it is always matched from the options array or pre-determined constant strings like config. getopt might match aggressively but ParseOption is called with controlled input